src/qt/networkstyle.cpp

```diff
@@ -20,78 +23,72 @@ static const struct {
+            col.getHsl(&h,&s,&l);
 
-            // loop through pixels
-            for(int x=0;x<img.width();x++)
+            // rotate color on RGB color circle
+            // 70° should end up with the typical "testnet" green
+            h+=iconColorHueShift;
+
+            // change saturation value
+            if (s>iconColorSaturationReduction)
             {
-                // preserve alpha because QColor::getHsl doesen't return the alpha value
-                a = qAlpha(scL[x]);
-                QColor col(scL[x]);
-
-                // get hue value
-                col.getHsl(&h,&s,&l);
-
-                // rotate color on RGB color circle
-                // 70° should end up with the typical "testnet" green
-                h+=iconColorHueShift;
-
-                // change saturation value
-                if(s>iconColorSaturationReduction)
-                {
-                    s -= iconColorSaturationReduction;
-                }
-                col.setHsl(h,s,l,a);
-
-                // set the pixel
-                scL[x] = col.rgba();
+                s -= iconColorSaturationReduction;
             }
-        }
+            col.setHsl(h,s,l,a);
 
-        //convert back to QPixmap
-#if QT_VERSION >= 0x040700
-        pixmap.convertFromImage(img);
-#else
-        pixmap = QPixmap::fromImage(img);
-#endif
+            // set the pixel
+            scL[x] = col.rgba();
+        }
     }
 
-    appIcon             = QIcon(pixmap);
-    trayAndWindowIcon   = QIcon(pixmap.scaled(QSize(256,256)));
+    return copy;
 }
 
-const NetworkStyle *NetworkStyle::instantiate(const QString &networkId)
+
+NetworkStyle::NetworkStyle(const QString &networkId)
 {
-    for (unsigned x=0; x<network_styles_count; ++x)
+    int iconColorHueShift = 0;
+    int iconColorSaturationReduction = 0;
+    // char *titleAddText;
+
+    for (int x=0; x<network_styles_count; ++x)
     {
         if (networkId == network_styles[x].networkId)
         {
-            return new NetworkStyle(
-                    network_styles[x].appName,
-                    network_styles[x].iconColorHueShift,
-                    network_styles[x].iconColorSaturationReduction,
-                    network_styles[x].titleAddText);
+            appName = network_styles[x].appName;
+            titleAddText = qApp->translate("SplashScreen", network_styles[x].titleAddText);
+            iconColorHueShift = network_styles[x].iconColorHueShift;
+            iconColorSaturationReduction = network_styles[x].iconColorSaturationReduction;
+            break;
         }
     }
-    return 0;
+    if (appName.isEmpty())
+        throw std::runtime_error("Unknown networkId passed into NetworkStyle");
+
+    appIcon = fixIcon(QImage(":/icons/bitcoin"), iconColorHueShift, iconColorSaturationReduction);
+    Q_ASSERT(appIcon.width() == 1024);
+    Q_ASSERT(appIcon.height() == 1024);
+    trayAndWindowIcon = QIcon(QPixmap::fromImage(appIcon.scaled(256, 256)));
```

Comment:
`QIcon` could be avoided by changing `void MacDockIconHandler::setIcon(const QIcon &icon)` to `void MacDockIconHandler::setIcon(const QPixmap &pixmap)` (but required a 128x128 icon otherwise it might be invisible [had similar issues recently]).